service/history/api/create_workflow_util.go

```diff
@@ -99,16 +99,32 @@ func NewWorkflowWithSignal(
 			return nil, err
 		}
 	}
-
+	requestEagerExecution := startRequest.StartRequest.GetRequestEagerExecution()
 	// Generate first workflow task event if not child WF and no first workflow task backoff
-	if err := GenerateFirstWorkflowTask(
+	scheduledEventID, err := GenerateFirstWorkflowTask(
 		newMutableState,
 		startRequest.ParentExecutionInfo,
 		startEvent,
-	); err != nil {
+		requestEagerExecution,
+	)
+	if err != nil {
 		return nil, err
 	}
 
+	// If first workflow task should back off (e.g. cron or workflow retry) a workflow task will not be scheduled.
+	if requestEagerExecution && scheduledEventID != 0 {
```

Comment:
nit: for readability use newMutableState.HasPendingWorkflowTask()?

Reply:
I'm not sure I find it more readable but I'm fine with making this change